tests/test_requests.py

```diff
@@ -351,6 +351,22 @@ def test_request_cookie_overrides_session_cookie(self, httpbin):
         # Session cookie should not be modified
         assert s.cookies['foo'] == 'bar'
 
+    def test_session_with_explicitly_none_cookies_takes_request(self, httpbin):
+        s = requests.session()
+        s.cookies = None
+        r = s.get(httpbin('cookies'), cookies={'foo': 'baz'})
+        assert r.json()['cookies']['foo'] == 'baz'
+        # Session cookie should still be None
+        assert s.cookies is None
+
+    def test_session_with_explicitly_none_cookies(self, httpbin):
+        s = requests.session()
+        s.cookies = None
+        r = s.get(httpbin('cookies'))
+        assert r.json()['cookies'] == {}
+        # Session cookie should still be None
+        assert s.cookies is None
```

Comment:
It'd be good to add a test where httpbin sets some cookies to prove that they also don't show up.